base/compiler/types.jl

```diff
@@ -82,20 +89,20 @@ is_removable_if_unused(effects::Effects) =
     effects.terminates === ALWAYS_TRUE &&
     effects.nothrow === ALWAYS_TRUE
 
-const EFFECTS_TOTAL = Effects(ALWAYS_TRUE, ALWAYS_TRUE, ALWAYS_TRUE, ALWAYS_TRUE)
-
 function encode_effects(e::Effects)
-    return e.consistent.state |
-           (e.effect_free.state << 2) |
-           (e.nothrow.state << 4) |
-           (e.terminates.state << 6)
+    return (e.consistent.state << 1) |
```

Comment:
This is bigger than a UInt8 now.

Reply:
Since `e.overlayed` is a `Bool`ean property, I think this can be encoded within `UInt8`?
```julia
julia> let maxbits = 0x02
           (maxbits << 1) |
           (maxbits << 3) |
           (maxbits << 5) |
           (maxbits << 7) |
           true
       end |> Int
85
```

Reply:
No, 4*2+1 is 9 bits. You're cutting of the top bit (in particular the top bit of the two bit unit you're shifting by 7).

Reply:
Au, you're right. Maybe we can pick `UInt16` instead though?

Reply:
I think UInt32 is fine. I have at least two more effects I think would be interesting, so with you're extra bit, and the one I already have in that PR, we'd be out again. It's not a huge amount of memory.